docs: mention support for multiple AD domains (#33273)
gravitational/teleport: docs/pages/desktop-access/active-directory-manual.mdx

Double click the "Add workstations to domain" policy and ensure that the
"Authenticated Users" group is not present.

> ## Multiple Domains

Author: Reminder to self: it probably makes sense to mention the version that this will be released in.

> ## Multiple Domains

Reviewer: ```suggestion
## Multiple domains
```
Using sentence case in headings

Each `windows_desktop_service` is designed to support connecting to hosts in a
single Active Directory domain. If you have multiple independent domains, you
can deploy multiple Teleport agents to service them.

If you have multiple domains with a trust relationship between them, you can
configure Teleport to perform PKI operations against a different domain.

In order for this to work, the hosts that you want to connect to and the AD

> users that you want to connect as must reside in the same domain.

Reviewer 2: > you can configure Teleport to perform PKI operations against a different domain.

> must reside in the same domain.

These two appear to contradict each other, consider whether there's a different and/or more precise way to word this.

Author: The users must be in the same domain as the computers. The PKI set up can be in a different domain. Open to suggestions for alternative ways to phrase it.

Reviewer 2: ```suggestion
Alternatively, if you have a pair of domains A and B with a trust relationship between them, you can
configure Teleport to perform PKI operations against domain A for users/hosts in domain B.

<Admonition type="warning" title="Supported Configuration">
  In order for this to work, the hosts that you want to connect to and the
  users that you want to connect as must reside in the same domain. Other
  configurations are not currently supported.
</Admonition>
```

Author: @ptgott what say you? I try to avoid fancy components and extra flair in favor of plain old text. Is this important enough to highlight?

Reviewer: I think this particular passage works just fine without the Admonition. The second paragraph flows naturally from the first, so (at least theoretically) users should move from the first paragraph to the second without the need for a box.

For example, suppose you have a root domain at `example.com` and a child domain
for developers at `dev.example.com`. If your PKI is configured at the root, but
you want to allow users in the child domain to connect to hosts in the child
domain, you would do the following:

> First, import Teleport's CA certificate as a trusted root certificate in the root

Reviewer: Since we're starting this list of instructions after a sentence ending in a colon, I think it would make visual sense to use an ordered list here.
```suggestion
1. Import Teleport's CA certificate as a trusted root certificate in the root
```

> domain's group policy and add the certificate to the NTAuth store as listed above.

Reviewer 2: ```suggestion
domain's group policy and add the certificate to the NTAuth store as described in the [section above](#publish-the-teleport-ca-to-the-ntauth-store).
```

> Next, configure Teleport to perform PKI against the root domain, while issuing

Reviewer: ```suggestion
1. Configure Teleport to perform PKI against the root domain, while issuing
```
Per the recommendation to use an ordered list above.

certificates for users and hosts in the child domain:

```yaml
windows_desktop_service:
  enabled: yes

  # configure LDAP settings to point at the child domain
  ldap:
    addr: dev.example.com:636
    username: 'DEV\svc-teleport'

  # optional: configure discovery for the child domain
  discovery:
    base_dn: CN=Computers,DC=dev,DC=example,DC=com

  # perform PKI against the root domain
  pki_domain: root.example.com
```

With this configuration, Teleport will generate certificates for users in
`dev.example.com`, but it will publish its CA and CRLs to `example.com`.

> 

Reviewer 2: Perhaps an `<Admonition type="warning" title="something">` warning that this won't work for other "types" of trust relationships, e.g. I'm inferring that it won't work if you have users at `another-example.com` because it's not a subdomain (though maybe that's incorrect).

Author: A subdomain is a DNS concept. I don't think it particularly matters what the name of the domain is so long as the trust is set up.
